tools/Microsoft.AspNetCore.Server.Kestrel.GeneratedCode/KnownHeaders.cs

```diff
@@ -411,12 +439,83 @@ protected void CopyToFast(ref MemoryPoolIterator output)
                                 output.CopyFromAscii(value);
                             }}
                         }}
+
+                    tempBits &= ~{1L << header.Index}L;
+                    if(tempBits == 0)
+                    {{
+                        return;
+                    }}
                 }}
             ")}
-        }}" : "")}
+
+            CopyToFastNonPrimary(ref output, tempBits);
+        }}
+        
+        protected void CopyToFastNonPrimary(ref MemoryPoolIterator output, long tempBits)
+        {{
+            {Each(loop.Headers.Where(h => !h.PrimaryHeader), header => $@"
+                if ({header.TestBit()})
+                {{ {(header.EnhancedSetter == false ? "" : $@"
+                    if (_headers._raw{header.Identifier} != null)
+                    {{
+                        output.CopyFrom(_headers._raw{header.Identifier}, 0, _headers._raw{header.Identifier}.Length);
+                    }}
+                    else ")}
+                        foreach (var value in _headers._{header.Identifier})
+                        {{
+                            if (value != null)
+                            {{
+                                output.CopyFrom(_headerBytes, {header.BytesOffset}, {header.BytesCount});
+                                output.CopyFromAscii(value);
+                            }}
+                        }}
+
+                    tempBits &= ~{1L << header.Index}L;
+                    if(tempBits == 0)
+                    {{
+                        return;
+                    }}
+                }}
+            ")}
+        }}
+        
+        " : "")}
         {(loop.ClassName == "FrameRequestHeaders" ? $@"
         public unsafe void Append(byte[] keyBytes, int keyOffset, int keyLength, string value)
         {{
+            fixed (byte* ptr = &keyBytes[keyOffset])
+            {{
+                var pUB = ptr;
+                var pUL = (ulong*)pUB;
+                var pUI = (uint*)pUB;
+                var pUS = (ushort*)pUB;
+                switch (keyLength)
+                {{{Each(loop.Headers.Where(h => h.PrimaryHeader).GroupBy(x => x.Name.Length), byLength => $@"
```

Comment:
Is this doing anything different than `AppendNonPrimaryHeaders`? Couldn't loop.Headers just be reordered to have put the PrimaryHeaders first?